### PR TITLE
build: remove use of TARGET_OS and BUILD_OS

### PR DESCRIPTION
**Pull request by fanquake**
In favour of checking `$host_os` and `$build_os` directly.

Guix Build (on x86_64):
```bash
68028f87229f37c1db16968cc7f573960e0b8807c907fcc2682224c318eed2ac  guix-build-86851c8e9424/output/aarch64-linux-gnu/SHA256SUMS.part
7eff4657c22ed71181286b753a65a5a4886a0e3ec3990dcd74961c0a5d33e9e0  guix-build-86851c8e9424/output/aarch64-linux-gnu/bitcoin-86851c8e9424-aarch64-linux-gnu-debug.tar.gz
5256ae19e65d9457862b37b68a7f12841c982822d21a03e3f45d0e461a3d3764  guix-build-86851c8e9424/output/aarch64-linux-gnu/bitcoin-86851c8e9424-aarch64-linux-gnu.tar.gz
15908fd7744342c66995d93a15cf08f27588dcf13bb199ce44d7805f50139161  guix-build-86851c8e9424/output/arm-linux-gnueabihf/SHA256SUMS.part
a77f6a0c53b664bdf7a5bd93bd9b7f983a7e9062d85ccae51c6828ccc88ed446  guix-build-86851c8e9424/output/arm-linux-gnueabihf/bitcoin-86851c8e9424-arm-linux-gnueabihf-debug.tar.gz
169d67890877bf9609ad0ac9d04d0651d6b3cb75b112c798f9bc9ac21c99a75b  guix-build-86851c8e9424/output/arm-linux-gnueabihf/bitcoin-86851c8e9424-arm-linux-gnueabihf.tar.gz
c3ef7422d577a34e23ac662dfe6269d7a8aa549cb723375f6e963ead3e2872a3  guix-build-86851c8e9424/output/arm64-apple-darwin/SHA256SUMS.part
a5dd82e4b151898473607b9796529f9e2021d252072b2c0321ddc87706f1ffde  guix-build-86851c8e9424/output/arm64-apple-darwin/bitcoin-86851c8e9424-arm64-apple-darwin-unsigned.dmg
75a8ff020808bbd3faad40c0760f1fef5bf30749e49f047ba13f9f43b9b09c9b  guix-build-86851c8e9424/output/arm64-apple-darwin/bitcoin-86851c8e9424-arm64-apple-darwin-unsigned.tar.gz
c7a0bd10d368be1373b105fab4eca2884a869def0e20d86f3e685a45a7759de9  guix-build-86851c8e9424/output/arm64-apple-darwin/bitcoin-86851c8e9424-arm64-apple-darwin.tar.gz
7fec22ad8ad1b9f6ec0014685ec46db216ae2c38040b2d9b02b6afbe0b726f8d  guix-build-86851c8e9424/output/dist-archive/bitcoin-86851c8e9424.tar.gz
6ad7faff882399eee3d9943fcaaabdae343cbf9e57e6bdd07af5a575735692ee  guix-build-86851c8e9424/output/powerpc64-linux-gnu/SHA256SUMS.part
2f39d53f910340bfb530f55d20c3658a2f4ff6ff854498c4ee81f4448f56595e  guix-build-86851c8e9424/output/powerpc64-linux-gnu/bitcoin-86851c8e9424-powerpc64-linux-gnu-debug.tar.gz
85071756fe637f1e69f68672b946ffd846cd2d940c3ff16e43710fb6cd8eb855  guix-build-86851c8e9424/output/powerpc64-linux-gnu/bitcoin-86851c8e9424-powerpc64-linux-gnu.tar.gz
99faa6d2cf4494ce04de65394a42e915739327fefbb474d162827988d41db97f  guix-build-86851c8e9424/output/powerpc64le-linux-gnu/SHA256SUMS.part
60e047bcfe417966a015b053c2ed866b072d2b2805dde8c9a34233fa2afc03bf  guix-build-86851c8e9424/output/powerpc64le-linux-gnu/bitcoin-86851c8e9424-powerpc64le-linux-gnu-debug.tar.gz
886d223badad0450516de403e308d550ea298ff240a2db5e1064d08f38af7a62  guix-build-86851c8e9424/output/powerpc64le-linux-gnu/bitcoin-86851c8e9424-powerpc64le-linux-gnu.tar.gz
02b68f0182a69f10e0b84438bba6fa4622a1e4f3d2eca9b9600caf3e8107b7ad  guix-build-86851c8e9424/output/riscv64-linux-gnu/SHA256SUMS.part
9a7121ad57093b5134c836367fd0f333b859e460374c5b3f5bcd50754d46b5ff  guix-build-86851c8e9424/output/riscv64-linux-gnu/bitcoin-86851c8e9424-riscv64-linux-gnu-debug.tar.gz
fbb814a427b5312063f1e136ad53728709e93647d62d927b200fe20e8794a8c4  guix-build-86851c8e9424/output/riscv64-linux-gnu/bitcoin-86851c8e9424-riscv64-linux-gnu.tar.gz
4b2b297264d1b18f574970b413571452e57ea2b7734b2e19b798db0651827dc5  guix-build-86851c8e9424/output/x86_64-apple-darwin/SHA256SUMS.part
db4e7273328d8a6643c4c96a1aef17b40e0e798ff42c4c68b8cea313d099f314  guix-build-86851c8e9424/output/x86_64-apple-darwin/bitcoin-86851c8e9424-x86_64-apple-darwin-unsigned.dmg
88835ac28bc88fa3f454ff771c18a6df18605cb046aa82f1e77e3e838d92a7cd  guix-build-86851c8e9424/output/x86_64-apple-darwin/bitcoin-86851c8e9424-x86_64-apple-darwin-unsigned.tar.gz
3b85604d469a18966743ef4d4711a800f1cd8ebba7cdc8d949348a60f5851815  guix-build-86851c8e9424/output/x86_64-apple-darwin/bitcoin-86851c8e9424-x86_64-apple-darwin.tar.gz
90850966a4c7a8fd44b14a95b29e6aaf5931ff00e78abe1623d79ac1d77a95ba  guix-build-86851c8e9424/output/x86_64-linux-gnu/SHA256SUMS.part
f960bfb2a7f2617aa530e96b03bda3ba89b4b39b4c6b1ff99bf2dbec2f6bcbcf  guix-build-86851c8e9424/output/x86_64-linux-gnu/bitcoin-86851c8e9424-x86_64-linux-gnu-debug.tar.gz
45bdb6ce51d9805f1ac6f11b7d61c4e8ef3c03afd2237904c16efefbd9130ba2  guix-build-86851c8e9424/output/x86_64-linux-gnu/bitcoin-86851c8e9424-x86_64-linux-gnu.tar.gz
2533b66a1fc86e36f371ce756c3c91b012cb8cf45c24fe8d06a0ffdaf44fdc79  guix-build-86851c8e9424/output/x86_64-w64-mingw32/SHA256SUMS.part
1666ce76ea847707932eed14eb9ac7ed9a64a86e33a5458a505e1ff7cf718d31  guix-build-86851c8e9424/output/x86_64-w64-mingw32/bitcoin-86851c8e9424-win64-debug.zip
21a18fa4eeaab9c3284894b2c4870ec1516cd5d013ef6580f9b2e60e1c0524a4  guix-build-86851c8e9424/output/x86_64-w64-mingw32/bitcoin-86851c8e9424-win64-setup-unsigned.exe
130d005feaf0e16928c5f7c10bc9d87cc3d713e7310ffdb5fcf633d6e017cfd5  guix-build-86851c8e9424/output/x86_64-w64-mingw32/bitcoin-86851c8e9424-win64-unsigned.tar.gz
1c0e77afa23541dccf27a850e00d8ecade82c95c70a58bb09499e43a61e48d44  guix-build-86851c8e9424/output/x86_64-w64-mingw32/bitcoin-86851c8e9424-win64.zip
```

### CODE REVIEW

**Comment by hebasto** [comment]
Concept ACK.

**Comment by fanquake** [comment]
Fixed the issue with the android build.

**Comment by fanquake** [comment]
Simplified this further and removed `TARGET_OS` and `BUILD_OS` entirely.

**Review by hebasto — COMMENTED** [review]
Approach ACK 3f84c220564d864328b76277fff30f1d1ae391fc

Why not being consistent about preferring `AS_CASE` over shell's `case`, at least within this PR?
See:
- https://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf-2.69/autoconf.html#Balancing-Parentheses
- https://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf-2.69/autoconf.html#case
- https://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf-2.69/autoconf.html#Common-Shell-Constructs

**Comment by fanquake** [comment]
Rebased past #24681.

**Comment by jarolrod** [comment]
Concept ACK

GUIX hashes on x86, mine match fanquake's latest round

```
find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum

68028f87229f37c1db16968cc7f573960e0b8807c907fcc2682224c318eed2ac  guix-build-86851c8e9424/output/aarch64-linux-gnu/SHA256SUMS.part
7eff4657c22ed71181286b753a65a5a4886a0e3ec3990dcd74961c0a5d33e9e0  guix-build-86851c8e9424/output/aarch64-linux-gnu/bitcoin-86851c8e9424-aarch64-linux-gnu-debug.tar.gz
5256ae19e65d9457862b37b68a7f12841c982822d21a03e3f45d0e461a3d3764  guix-build-86851c8e9424/output/aarch64-linux-gnu/bitcoin-86851c8e9424-aarch64-linux-gnu.tar.gz
15908fd7744342c66995d93a15cf08f27588dcf13bb199ce44d7805f50139161  guix-build-86851c8e9424/output/arm-linux-gnueabihf/SHA256SUMS.part
a77f6a0c53b664bdf7a5bd93bd9b7f983a7e9062d85ccae51c6828ccc88ed446  guix-build-86851c8e9424/output/arm-linux-gnueabihf/bitcoin-86851c8e9424-arm-linux-gnueabihf-debug.tar.gz
169d67890877bf9609ad0ac9d04d0651d6b3cb75b112c798f9bc9ac21c99a75b  guix-build-86851c8e9424/output/arm-linux-gnueabihf/bitcoin-86851c8e9424-arm-linux-gnueabihf.tar.gz
c3ef7422d577a34e23ac662dfe6269d7a8aa549cb723375f6e963ead3e2872a3  guix-build-86851c8e9424/output/arm64-apple-darwin/SHA256SUMS.part
a5dd82e4b151898473607b9796529f9e2021d252072b2c0321ddc87706f1ffde  guix-build-86851c8e9424/output/arm64-apple-darwin/bitcoin-86851c8e9424-arm64-apple-darwin-unsigned.dmg
75a8ff020808bbd3faad40c0760f1fef5bf30749e49f047ba13f9f43b9b09c9b  guix-build-86851c8e9424/output/arm64-apple-darwin/bitcoin-86851c8e9424-arm64-apple-darwin-unsigned.tar.gz
c7a0bd10d368be1373b105fab4eca2884a869def0e20d86f3e685a45a7759de9  guix-build-86851c8e9424/output/arm64-apple-darwin/bitcoin-86851c8e9424-arm64-apple-darwin.tar.gz
7fec22ad8ad1b9f6ec0014685ec46db216ae2c38040b2d9b02b6afbe0b726f8d  guix-build-86851c8e9424/output/dist-archive/bitcoin-86851c8e9424.tar.gz
6ad7faff882399eee3d9943fcaaabdae343cbf9e57e6bdd07af5a575735692ee  guix-build-86851c8e9424/output/powerpc64-linux-gnu/SHA256SUMS.part
2f39d53f910340bfb530f55d20c3658a2f4ff6ff854498c4ee81f4448f56595e  guix-build-86851c8e9424/output/powerpc64-linux-gnu/bitcoin-86851c8e9424-powerpc64-linux-gnu-debug.tar.gz
85071756fe637f1e69f68672b946ffd846cd2d940c3ff16e43710fb6cd8eb855  guix-build-86851c8e9424/output/powerpc64-linux-gnu/bitcoin-86851c8e9424-powerpc64-linux-gnu.tar.gz
99faa6d2cf4494ce04de65394a42e915739327fefbb474d162827988d41db97f  guix-build-86851c8e9424/output/powerpc64le-linux-gnu/SHA256SUMS.part
60e047bcfe417966a015b053c2ed866b072d2b2805dde8c9a34233fa2afc03bf  guix-build-86851c8e9424/output/powerpc64le-linux-gnu/bitcoin-86851c8e9424-powerpc64le-linux-gnu-debug.tar.gz
886d223badad0450516de403e308d550ea298ff240a2db5e1064d08f38af7a62  guix-build-86851c8e9424/output/powerpc64le-linux-gnu/bitcoin-86851c8e9424-powerpc64le-linux-gnu.tar.gz
02b68f0182a69f10e0b84438bba6fa4622a1e4f3d2eca9b9600caf3e8107b7ad  guix-build-86851c8e9424/output/riscv64-linux-gnu/SHA256SUMS.part
9a7121ad57093b5134c836367fd0f333b859e460374c5b3f5bcd50754d46b5ff  guix-build-86851c8e9424/output/riscv64-linux-gnu/bitcoin-86851c8e9424-riscv64-linux-gnu-debug.tar.gz
fbb814a427b5312063f1e136ad53728709e93647d62d927b200fe20e8794a8c4  guix-build-86851c8e9424/output/riscv64-linux-gnu/bitcoin-86851c8e9424-riscv64-linux-gnu.tar.gz
4b2b297264d1b18f574970b413571452e57ea2b7734b2e19b798db0651827dc5  guix-build-86851c8e9424/output/x86_64-apple-darwin/SHA256SUMS.part
db4e7273328d8a6643c4c96a1aef17b40e0e798ff42c4c68b8cea313d099f314  guix-build-86851c8e9424/output/x86_64-apple-darwin/bitcoin-86851c8e9424-x86_64-apple-darwin-unsigned.dmg
88835ac28bc88fa3f454ff771c18a6df18605cb046aa82f1e77e3e838d92a7cd  guix-build-86851c8e9424/output/x86_64-apple-darwin/bitcoin-86851c8e9424-x86_64-apple-darwin-unsigned.tar.gz
3b85604d469a18966743ef4d4711a800f1cd8ebba7cdc8d949348a60f5851815  guix-build-86851c8e9424/output/x86_64-apple-darwin/bitcoin-86851c8e9424-x86_64-apple-darwin.tar.gz
90850966a4c7a8fd44b14a95b29e6aaf5931ff00e78abe1623d79ac1d77a95ba  guix-build-86851c8e9424/output/x86_64-linux-gnu/SHA256SUMS.part
f960bfb2a7f2617aa530e96b03bda3ba89b4b39b4c6b1ff99bf2dbec2f6bcbcf  guix-build-86851c8e9424/output/x86_64-linux-gnu/bitcoin-86851c8e9424-x86_64-linux-gnu-debug.tar.gz
45bdb6ce51d9805f1ac6f11b7d61c4e8ef3c03afd2237904c16efefbd9130ba2  guix-build-86851c8e9424/output/x86_64-linux-gnu/bitcoin-86851c8e9424-x86_64-linux-gnu.tar.gz
2533b66a1fc86e36f371ce756c3c91b012cb8cf45c24fe8d06a0ffdaf44fdc79  guix-build-86851c8e9424/output/x86_64-w64-mingw32/SHA256SUMS.part
1666ce76ea847707932eed14eb9ac7ed9a64a86e33a5458a505e1ff7cf718d31  guix-build-86851c8e9424/output/x86_64-w64-mingw32/bitcoin-86851c8e9424-win64-debug.zip
21a18fa4eeaab9c3284894b2c4870ec1516cd5d013ef6580f9b2e60e1c0524a4  guix-build-86851c8e9424/output/x86_64-w64-mingw32/bitcoin-86851c8e9424-win64-setup-unsigned.exe
130d005feaf0e16928c5f7c10bc9d87cc3d713e7310ffdb5fcf633d6e017cfd5  guix-build-86851c8e9424/output/x86_64-w64-mingw32/bitcoin-86851c8e9424-win64-unsigned.tar.gz
1c0e77afa23541dccf27a850e00d8ecade82c95c70a58bb09499e43a61e48d44  guix-build-86851c8e9424/output/x86_64-w64-mingw32/bitcoin-86851c8e9424-win64.zip

```

File: configure.ac [hunk 1]
```diff
@@ -1749,9 +1724,7 @@ else
     fi
     AC_MSG_RESULT($use_natpmp_default)
     AC_DEFINE_UNQUOTED([USE_NATPMP], [$natpmp_setting], [NAT-PMP support not compiled if undefined, otherwise value (0 or 1) determines default state])
-    if test "$TARGET_OS" = "windows"; then
-      NATPMP_CPPFLAGS="-DSTATICLIB -DNATPMP_STATICLIB"
-    fi
+    AS_CASE([$host_os], [*mingw*], [NATPMP_CPPFLAGS="-DSTATICLIB -DNATPMP_STATICLIB"])
```
Comment (laanwj) [inline configure.ac:1727]: What about non-mingw windows? Does this make it more difficult to switch compilers later? E.g. does the clang windows tuple have mingw in it?

**Pull request closed without merging**